hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -523,12 +530,19 @@ private ClientDisconnectionOperation createClientDisconnectionOperation(String c
 
         @Override
         public void run() {
+            InternalOperationService service = nodeEngine.getOperationService();
+            Address thisAddr = getLocalMember().getAddress();
             for (Map.Entry<String, String> entry : ownershipMappings.entrySet()) {
                 String clientUuid = entry.getKey();
                 String memberUuid = entry.getValue();
                 if (deadMemberUuid.equals(memberUuid)) {
-                    ClientDisconnectionOperation op = createClientDisconnectionOperation(clientUuid, deadMemberUuid);
-                    nodeEngine.getOperationService().run(op);
+                    ClientDisconnectionOperation op = new ClientDisconnectionOperation(clientUuid, memberUuid);
+                    Future future = service.createInvocationBuilder(ClientEngineImpl.SERVICE_NAME, op, thisAddr).invoke();
```

Comment:
What's the reason of doing a local & blocking invocation on a generic thread?

Reply:
Actual change here was to convert it to invocation builder so that it can be retried properly. 
I can remove back the blocking part, that does not seem necessary anymore.